nexus_constructor/InstrumentView.py

```diff
@@ -0,0 +1,260 @@
+from PySide2.QtWidgets import QWidget, QVBoxLayout
+from PySide2.Qt3DExtras import Qt3DExtras
+from PySide2.Qt3DCore import Qt3DCore
+from PySide2.QtCore import QPropertyAnimation
+from PySide2.QtGui import QVector3D, QColor, QMatrix4x4
+from nexus_constructor.NeutronAnimationController import NeutronAnimationController
+
+
+class InstrumentView(QWidget):
+    """
+    Class for managing the 3D view in the NeXus Constructor. Creates the initial sample, the initial beam, and the
+    neutron animation.
+    :param parent: The MainWindow in which this widget is created. This isn't used for anything but is accepted as an
+                   argument in order to appease Qt Designer.
+    """
+
+    def __init__(self, parent):
+        super().__init__()
+        lay = QVBoxLayout(self)
+        self.view = Qt3DExtras.Qt3DWindow()
+        self.view.defaultFrameGraph().setClearColor(QColor("lightgrey"))
+        container = QWidget.createWindowContainer(self.view)
+        lay.addWidget(container)
+
+        self.view.camera().lens().setPerspectiveProjection(45, 16 / 9, 0.1, 1000)
```

Comment:
maybe comments here describing why we have these magic numbers that set the position etc. 

explain why they're set to what they are

Reply:
I'll add something although this was copied from QML so I don't entirely understand it myself.